File: source/com/barrybecker4/simulation/liquid/config/Region.java
/** Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT  */
package com.barrybecker4.simulation.liquid.config;

import com.barrybecker4.common.geometry.Location;

/**
 * Represents a region of something (like a source sink, or liquid block) in the simulation.
 *
 * @author Barry Becker
 */
public class Region {

    /** starting location for the rectangular region */
    Location start;

    /** optional stopping location for the rectangular region */
    Location stop;

    /**
     * Constructor
     */
    public Region(Location start, Location stop) {
        this.start = start;
        this.stop = stop;
    }

    public Location getStart() {
        return start;
    }

    public Location getStop() {
        if (stop == null) {
            return start;
        }
        return stop;
    }

}

File: scala-source/com/barrybecker4/simulation/liquid/config/Region.scala
// Copyright by Barry G. Becker, 2016-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.simulation.liquid.config

import com.barrybecker4.common.geometry.Location


/**
  * Represents a region of something (like a source sink, or liquid block) in the simulation.
  * @param start starting location for the rectangular region
  * @param stop optional stopping location for the rectangular region
  * @author Barry Becker
  */
class Region(var start: Location, var stop: Location) {
  def getStart: Location = start
  def getStop: Location = { if (stop == null)  start else stop }
}
